lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1837,6 +1837,42 @@ bool QuakeBridgeVisitor::VisitInitListExpr(clang::InitListExpr *x) {
     // Pass initialization list with one member as a Ref.
     return pushValue(last[0]);
   }
+
+  // Check if all values here are Integer or Float Type
+  bool isAllIntOrAllFloat = [&]() {
+    for (auto v : last)
+      if (!v.getType().isIntOrFloat())
+        return false;
+    return true;
+  }();
```

Comment:
I'm curious - is there a benefit to doing this as a lamba vs just initializing `isAllIntOrAllFloat = true`, and then iterating through the loop and setting `isAllIntOrAllFloat = false` if you find out otherwise? From a stepping-through-a-debugger perspective, the latter sounds easier.

Reply:
I like the functional style since it makes it clear that the variable is properly initialized where it is declared. I'm not sure what the debugger issue is; I can and do step through lambda calls.

Reply:
The debugger issue isn't a showstopper by any means. I just have to remember to F11 into it instead of F10. It sounds like it is somewhat personal/style preference, which is definitely fine by me. 👍